README.adoc

```diff
@@ -0,0 +1,402 @@
+:toc: macro
+:toc-title:
+:toclevels: 99
+
+# SSENSE JavaScript Style Guide
+
+toc::[]
+
+
+
+## Installation
+```
+npm install --save-dev eslint eslint-config-ssense
+```
+
+Because of the https://github.com/eslint/eslint/issues/3458[current inability for sharable configs] to supply their dependencies you will also need to:
+
+```
+npm install --save-dev \
+  babel-eslint \
+  eslint-config-airbnb-base \
+  eslint-plugin-import \
+  eslint-plugin-fp \
+  eslint-plugin-vue
+```
+
+
+
+## Usage
+
+In `package.json`:
+
+```json
+  "eslintConfig": {
+    "extends": "ssense"
+  }
+```
+
+(Defaults to `server`)
+
+```json
+  "eslintConfig": {
+    "extends": "ssense/server"
+  }
+```
+```json
+  "eslintConfig": {
+    "extends": "ssense/client"
+  }
+```
+
+There are two specialized configurations available, `client` or `server`. `client` requires the `vue` plugin and does not regulate `this` usage because Vue relies on it heavily (unfortunately). Unless you are writing Vue code do not use `client`. By default `server` is used.
+
+
+
+
+## Plugins
+
+#### `fp` (https://github.com/jfmengels/eslint-plugin-fp[↗])
+Provides rules that help enforce functional programming.
+
+**Talks around FP**
+
+* https://www.infoq.com/presentations/Value-Values[The Value of Values] – Rich Hickey
+* https://www.infoq.com/presentations/Are-We-There-Yet-Rich-Hickey[Are we there yet?] – Rich Hickey
+* https://www.youtube.com/watch?v=DMtwq3QtddY[The Functional Final Frontier] – David Nolen
+* https://www.youtube.com/watch?v=mS264h8KGwk[Immutability, interactivity & JavaScript] – David Nolen
+
+**Writings around FP**
+
+* https://github.com/stoeffel/awesome-fp-js[Awesome FP JS] list
+* http://blog.wolksoftware.com/the-rise-of-functional-programming-and-the-death-of-angularjs[The rise of functional programming & the decline of Angular 2.0]
+
+#### `import` (https://github.com/benmosher/eslint-plugin-import[↗])
+Provides rules that help prevent import bugs and enforces style.
+
+#### `vue` (https://github.com/vuejs/eslint-plugin-vue[↗])
+Currently, the sole purpose of having this plugin is to parse https://vuejs.org/v2/guide/single-file-components.html[`.vue` files]. Also, it https://github.com/vuejs/eslint-plugin-vue/issues/1[does not support] eslint's `--fix` feature.
+
+## Extends
+
+#### Rules of AirBnB (https://github.com/airbnb/javascript[↗])
+
+We extend the AirBnB rules for historical reasons. Our configuration will continue to evolve and may not be based on it one day if we eventually disable or adjust too much of it via overrides.
+
+
+
+
+## Rules
+
+### `semi`
```

Comment:
Titles currently correspond to their eslint rule names.  In this case it's a little lame. What is your alternative exactly? We could write humane titles with their rule name following? Will try something out.